src/ExadelMentorship.WebApi/ExceptionHandler.cs

```diff
@@ -8,20 +11,33 @@ public static ExceptionHandlerOptions GetExceptionHandlerOptions()
         {
             return new ExceptionHandlerOptions()
             {
-                ExceptionHandler = (c) =>
+                ExceptionHandler = async (c) =>
                 {
                     var exception = c.Features.Get<IExceptionHandlerFeature>();
-                    var statusCode = exception?.Error.GetType().Name switch
+                    if (exception is not null)
                     {
-                        "NotFoundException" => StatusCodes.Status404NotFound,
-                        "FormatException" => StatusCodes.Status400BadRequest,
-                        _ => StatusCodes.Status500InternalServerError
-                    };
-                    c.Response.StatusCode = statusCode;
-                    var message = exception is not null ? exception.Error.Message : String.Empty;
-                    c.Response.WriteAsync(message);
-
-                    return Task.CompletedTask;
+                        int statusCode;
+                        string message = string.Empty;
+                        switch (exception?.Error)
+                        {
+                            case NotFoundException:
+                                statusCode = StatusCodes.Status404NotFound;
+                                message = exception.Error.Message;
+                                break;
+                            case ValidationException:
+                                statusCode = StatusCodes.Status400BadRequest;
+                                var validationException = (ValidationException)exception.Error;
+                                var err = validationException.Errors;
+                                await c.Response.WriteAsync(err.ToString());
```

Comment:
Why not pass `err` itself instead of `.ToString()`? Errors is a list, we can pass it as a JSON, instead of plain text.

Reply:
done